### PR TITLE
PR: Do not close split editor when empty or when switching project

### LINKED ISSUE #5898: Closing all files in split pane view with at least 2 open then clicking any menu triggers RuntimeError
## Description

**What steps will reproduce the problem?**

1. 
2. 
3. 

**What is the expected output? What do you see instead?**

**Please provide any additional information below**

  File "/home/dhama/anaconda3/lib/python3.6/site-packages/spyder/app/mainwindow.py", line 2013, in update_edit_menu
    widget, textedit_properties = self.get_focus_widget_properties()
  File "/home/dhama/anaconda3/lib/python3.6/site-packages/spyder/app/mainwindow.py", line 2006, in get_focus_widget_properties
    not_readonly = not widget.isReadOnly()
RuntimeError: wrapped C/C++ object of type CodeEditor has been deleted

## Version and main components

* Spyder Version: 3.2.4 
* Python Version: 3.6.3
* Qt Versions: 5.6.2, PyQt5 5.6 on Linux

## Dependencies
```
pyflakes >=0.6.0 :  1.6.0 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pygments >=2.0   :  2.2.0 (OK)
pandas >=0.13.1  :  0.20.3 (OK)
numpy >=1.7      :  1.13.3 (OK)
sphinx >=0.6.6   :  1.6.3 (OK)
rope >=0.9.4     :  0.10.5 (OK)
jedi >=0.9.0     :  0.10.2 (OK)
psutil >=0.3     :  5.4.0 (OK)
nbconvert >=4.0  :  5.3.1 (OK)
sympy >=0.7.3    :  1.1.1 (OK)
cython >=0.21    :  0.26.1 (OK)
qtconsole >=4.2.0:  4.3.1 (OK)
IPython >=4.0    :  6.1.0 (OK)
pylint >=0.25    :  1.7.4 (OK)

```

### PR DESCRIPTION
**Pull request by jnsebgosselin**
### Pull Request Checklist

* [x] Read and followed this repo's [Contributing Guidelines](https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md)
* [x] Based this PR on the latest version of the correct branch (master or 3.x)
* [x] Followed [PEP 8](https://www.python.org/dev/peps/pep-0008/) code style
* [x] Ensured this pull request hasn't eliminated unrelated blank lines/spaces,
      modified the ``spyder/defaults`` directory, or added new icons/assets
* [x] Wrote at least one-line docstrings following PEP 257for any new functions
* [ ] Added at least one unit test covering the changes, if at all possible
* [x] Described the changes and the motivation for them below
* [x] Noted what issue(s) this pull request resolves, creating one if needed
* [ ] Included a screenshot, if this PR makes any visible changes to the UI

### Developer Certificate of Origin Affirmation

By submitting this Pull Request or typing my name below, I affirm the
[Developer Certificate of Origin](https://developercertificate.org/)
with respect to both the content of the contribution itself and this post,
and understand I am releasing it under Spyder's MIT (Expat) license.

## Description of Changes

- Do not close an editorstack when closing the last file in it.
- Setup the split and window layout only the first time files are setup in the editor, instead of everytime.

### Issue(s) Resolved

Fixes  #5898

### CODE REVIEW

File: spyder/plugins/editor.py [hunk 1]
```diff
@@ -2690,30 +2687,31 @@ def set_open_filenames(self):
...
- 
+
         if filenames and any([osp.isfile(f) for f in filenames]):
             self.load(filenames)
-            layout = self.get_option('layout_settings', None)
-            if layout is not None:
-                self.editorsplitter.set_layout_settings(layout)
-            win_layout = self.get_option('windows_layout_settings', None)
-            if win_layout:
+            if self.__first_open_files_setup:
```
Comment (ccordoba12) [inline spyder/plugins/editor.py]: So this only applies the first time an Editorstack is created?

Comment (jnsebgosselin) [reply spyder/plugins/editor.py]: Yes, that is the idea. Do you think there is corner cases where this can cause some problems?

Comment (ccordoba12) [reply spyder/plugins/editor.py]: No, or at least I can't think of any right now. But that's why it'd be good to have a test for this.

Comment (jnsebgosselin) [reply spyder/plugins/editor.py]: Ok, I'll see what I can do for the test.

**Comment by jnsebgosselin** [comment]
Thanks Carlos and sorry for not adding a test in time.
